File: sdk/testing/sl4f/client/lib/src/time.dart
// Copyright 2020 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import 'dart:core';

import 'sl4f_client.dart';

/// Check the time as reported on the DUT using [Sl4f].
class Time {
  final Sl4f _sl4f;

  /// Constructs a [Time] object.
  Time(this._sl4f);

  /// Returns the system time on the DUT in UTC.
  Future<DateTime> systemTime() async {
    final timestampMillis = await _sl4f.request('time_facade.SystemTimeMillis');
    return DateTime.fromMillisecondsSinceEpoch(timestampMillis, isUtc: true);
  }

  /// Returns whether or not system time on the DUT is synchronized to an external source.
  Future<bool> isSystemTimeSynchronized() async =>
      await _sl4f.request('time_facade.IsSynchronized');
}

File: src/diagnostics/transit_bench/src/time.rs
// Copyright 2019 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

use {
    fuchsia_criterion::criterion::{black_box, Bencher, Criterion},
    fuchsia_zircon as zx,
};

pub fn benches(c: &mut Criterion) {
    c.bench_function("ticks_get", ticks_get);
    c.bench_function("monotonic_time", monotonic_time);
}

fn ticks_get(b: &mut Bencher) {
    b.iter(|| black_box(zx::ticks_get()));
}

fn monotonic_time(b: &mut Bencher) {
    b.iter(|| black_box(zx::Time::get_monotonic()));
}
